### PR TITLE
Parser and AST working

### PR DESCRIPTION
**Pull request by christopher-jensen**

### CODE REVIEW

File: AntlrB/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Datasource local storage ignored files
+/../../../../../../../:\Users\chris\4-semester\Projekt\AntlrB\.idea/dataSources/
+/dataSources.local.xml
+# Editor-based HTTP Client requests
+/httpRequests/
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/.idea/.gitignore]: Delete this

File: AntlrB/src/B/B.g4 [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                  ;
+funcdefparam : type ID ;
+stmts : IF LPAREN expr RPAREN RETURN stmt stmts                               #stmtsNotEmpty
+      |                                                                       #stmtsEmpty
+      ;
+stmt : expr SEMICOLON ;
+expr : left=logexpr op=LOGOR right=expr                                       #logOr
+     | logexpr                                                                #log // logexp
+     ;
+logexpr : left=logexpr2 op=LOGAND right=logexpr                               #logAnd
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/src/B/B.g4]:
```suggestion
logexpr : left=logexpr2 op=LOGAND right=logexpr                               #logAndOp
```

File: AntlrB/src/B/B.g4 [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+              ;
+funcdefmoreparams : COMMA funcdefparam funcdefmoreparams                      #funcdefmoreparamsNotEmpty //dclMoreParams
+                  |                                                           #funcdefmoreparamsEmpty
+                  ;
+funcdefparam : type ID ;
+stmts : IF LPAREN expr RPAREN RETURN stmt stmts                               #stmtsNotEmpty
+      |                                                                       #stmtsEmpty
+      ;
+stmt : expr SEMICOLON ;
+expr : left=logexpr op=LOGOR right=expr                                       #logOr
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/src/B/B.g4]:
```suggestion
expr : left=logexpr op=LOGOR right=expr                                       #logOrOp
```

File: AntlrB/src/B/Nodes/Abstract/Code.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package B.Nodes.Abstract;
+
+public abstract class Code implements Node{
+
+}
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/src/B/Nodes/Abstract/Code.java]: Delete this file

File: AntlrB/src/B/Nodes/Nodes/CodeFuncDef.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package B.Nodes.Nodes;
+
+import B.Nodes.Abstract.*;
+
+public class CodeFuncDef extends Code {
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/src/B/Nodes/Nodes/CodeFuncDef.java]:
```suggestion
package B.Nodes.Nodes;

import B.Nodes.Abstract.*;

public class CodeFuncDef extends Node {
```

File: AntlrB/src/B/Nodes/Nodes/CodeStmt.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package B.Nodes.Nodes;
+
+import B.Nodes.Abstract.*;
+
+public class CodeStmt extends Code {
```
Comment (JacobToftgaardRasmussen) [inline AntlrB/src/B/Nodes/Nodes/CodeStmt.java]:
```suggestion
package B.Nodes.Nodes;

import B.Nodes.Abstract.*;

public class CodeStmt extends Node {
```

File: CUP-parser/AST/Nodes/Followterm.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+public class Followterm implements Node {
+    public Followterm(Simpleops simpleops, Math math) {
+        Simpleops = simpleops;
+        this.Math = math;
+    }
+
+    public String NodeType = "Followterm";
+    public Simpleops Simpleops;
+    public Math Math;
+}
```
Comment (JacobToftgaardRasmussen) [inline CUP-parser/AST/Nodes/Followterm.java:14]: Delete this

**Review by JacobToftgaardRasmussen — COMMENTED** [review]
NotType property should be removed from all nodes

File: AntlrB/.idea/.gitignore [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Datasource local storage ignored files
+/../../../../../../../:\Users\chris\4-semester\Projekt\AntlrB\.idea/dataSources/
```
Comment (humleflue) [inline AntlrB/.idea/.gitignore]: This looks wrong...

**Review by humleflue — CHANGES_REQUESTED** [review]
Looks good over all!

I'd like the root folder's name to be renamed to something like ANTLR-AST instead of AntlrB.
Altso look at all the comments Jacob made as well as the one I made.

**Review by humleflue — APPROVED** [review]
LGTM! :trollface:

**Pull request merged**